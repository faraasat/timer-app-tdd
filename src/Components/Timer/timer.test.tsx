import { EnzymePropSelector, shallow } from "enzyme";
import TimerComponent from "./timer.component";

describe("Timer Component Shallow Render", () => {
  let shallowRender: EnzymePropSelector;
  beforeEach(() => {
    shallowRender = shallow(<TimerComponent />);
  });

  it("should match snapshot", () => {
    expect(shallowRender.debug()).toMatchSnapshot();
  });

  it("must have one div", () => {
    expect(shallowRender.find("div").length).toBeGreaterThanOrEqual(2);
  });

  it("should have class timer-component-root", () => {
    expect(shallowRender.find(".timer-component-root").length).toEqual(1);
  });

  it("should have class timer-component-alignment", () => {
    expect(shallowRender.find(".timer-component-alignment").length).toEqual(1);
  });
});
