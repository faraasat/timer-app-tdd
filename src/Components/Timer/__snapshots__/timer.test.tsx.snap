// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer Component Shallow Render should match snapshot 1`] = `
"<div className=\\"timer-component-root\\">
  <TimerScreenComponent playState={false} pauseState={false} resetState={false} setResetState={[Function: bound _dispatchAction]} />
  <div className=\\"timer-component-alignment\\">
    <TimerButtonComponent id=\\"start-timer\\" onClick={[Function: onClick]} toggle={false} IconComponent={{...}} color=\\"green\\" />
    <TimerButtonComponent id=\\"stop-timer\\" onClick={[Function: onClick]} toggle={false} IconComponent={{...}} color=\\"red\\" />
    <TimerButtonComponent id=\\"pause-timer\\" onClick={[Function: onClick]} toggle={false} IconComponent={{...}} color=\\"blue\\" />
  </div>
</div>"
`;
